src/injectHighlightWrappers.spec.ts

```diff
@@ -0,0 +1,357 @@
+import Highlight from './Highlight';
+import injectHighlightWrappers, { DATA_ATTR } from './injectHighlightWrappers';
+import { adjacentTextSections, imageBetweenParagraphs, paragraphFigureAndCaption } from './injectHighlightWrappers.spec.data';
+
+describe('inject highlight wrappers for figure with caption', () => {
+  let page: HTMLElement;
+  let img: HTMLElement;
+  let p: HTMLElement;
+  let textNode: Node;
+  let span: HTMLElement;
+  let captionTitle: HTMLElement;
+  let captionContainer: HTMLElement;
+  let captionTitleText: Node;
+  let rangeDefaults: {};
+
+  const highlightData = { id: 'some-highlight', content: 'asd', style: 'yellow' };
+
+  beforeEach(() => {
+    document.body.innerHTML = paragraphFigureAndCaption;
+    page = document.getElementById('container')!;
+    img = document.getElementById('test-img')!;
+    p = document.getElementById('test-p')!;
+    textNode = p.childNodes[0];
+    span = document.getElementById('test-span')!;
+    captionContainer = document.getElementById('test-caption-container')!;
+    captionTitle = document.getElementById('test-caption-title')!;
+    captionTitleText = captionTitle.childNodes[0];
+
+    Date.now = jest.fn();
+
+    rangeDefaults = {
+      collapse: false,
+      commonAncestorContainer: page,
+      setEndAfter: jest.fn(),
+      setStartBefore: jest.fn(),
+    };
+  });
+
+  describe('for highlight ending on an <img>', () => {
+
+    it('in chrome and safari', () => {
+      const range: any = {
+        ...rangeDefaults,
+        endContainer: span,
+        endOffset: 2,
+        startContainer: textNode,
+        startOffset: 2,
+      };
+      const highlight = new Highlight(range, highlightData, { formatMessage: jest.fn() });
+
+      injectHighlightWrappers(highlight);
+      const highlightSpans = document.querySelectorAll(`[${DATA_ATTR}='true']`);
+
+      highlightSpans.forEach((el) => {
+        expect(el).toMatchSnapshot();
+      });
+    });
+
+    it('in firefox', () => {
+      const range: any = {
+        ...rangeDefaults,
+        endContainer: captionTitleText,
+        endOffset: 0,
+        startContainer: textNode,
+        startOffset: 2,
+      };
+
+      const highlight = new Highlight(range, highlightData, { formatMessage: jest.fn() });
+
+      injectHighlightWrappers(highlight);
+      const highlightSpans = document.querySelectorAll(`[${DATA_ATTR}='true']`);
+
+      highlightSpans.forEach((el) => {
+        expect(el).toMatchSnapshot();
+      });
+    });
+  });
+
+  describe('for highlight starting on an <img>', () => {
+
+    it('in chrome and safari', () => {
+      const range: any = {
+        ...rangeDefaults,
+        endContainer: captionTitleText,
+        endOffset: 6,
+        startContainer: textNode,
+        startOffset: textNode.nodeValue!.length,
+      };
+
+      const highlight = new Highlight(range, highlightData, { formatMessage: jest.fn() });
```

Comment:
maybe mock the formatMessage implementation so that we get something in the snapshot other than `undefined`

Reply:
Updated to mock formatMessage and to create a highlight instance for each range instead of just injecting wrappers (this way the `highlight` class is added rather than `undefined`)